zt_backend/utils/linting.py

```diff
@@ -0,0 +1,150 @@
+import asyncio
+import json
+from typing import Dict, List
+import logging
+from collections import defaultdict
+import time
+
+# Constants
+DEBOUNCE_TIME = 0.5  # seconds
+RUFF_COMMAND = ['ruff', 'check', '--output-format=json', '-']
+
+# Setup logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Debounce settings
+last_run_time = defaultdict(float)
+pending_tasks: Dict[str, asyncio.Task] = {}
+
+async def run_ruff_linting(text: str) -> List[Dict]:
+    process = await asyncio.create_subprocess_exec(
+        *RUFF_COMMAND,
```

Comment:
Make sure to update the setup.cfg with this new dependency. Is there no way to do this programmatically rather than CLI?

Reply:
@Carson-Shaar , Ruff is a Command-line tool, I did checked on that again but, we need to use commands only as they dont provide anything else